spinalcordtoolbox/reports/qc2.py

```diff
@@ -1018,18 +1018,30 @@ def assign_label_colors_by_groups(labels):
     return color_list
 
 
-def crop_with_mask(array, img_crop, pad=3, axis=0):
+def crop_with_mask(array, img_crop, pad=3, axis=0, total_slices=None):
     """
     Crop array along a specific axis based on nonzero slices in the reference image.
+
+    Use `pad` if you want to pad around the mask (no matter how big the mask is). Use
+    `total_slices` to pad only until the total number of slices is reached (overrides `pad`).
+
     Note: We use axis=0 by default because QC images are reoriented to SAL, therefore we
         crop the SI axis by default.
     """
     # get extents of segmentation used for cropping
     first_slice = min(np.where(img_crop.data)[axis])
     last_slice = max(np.where(img_crop.data)[axis])
+    # if `total_slices` is specified, then override `pad`
+    if total_slices is not None:
+        pad_total = max(0, total_slices - (last_slice - first_slice + 1))
+        l_pad = math.floor(pad_total // 2)
+        r_pad = math.ceil(pad_total // 2)
```

Comment:
Whoops, this is integer division, I meant to do float division here (with floor/ceil).

```suggestion
        l_pad = math.floor(pad_total / 2)
        r_pad = math.ceil(pad_total / 2)
```